### PR TITLE
Update general rules of masthead menu

### LINKED ISSUE #570: Inconsistent story for masthead icons, pop-ups, dropdowns
I would like to report a bug on consistent use of a pattern in our design documentation and visuals in the masthead.

Today, we show a caret next to the help dropdown, but not the launcher popup.  The fact that these two icons in the masthead act differently, one as a dropdown and one as a pop-up, seems inconsistent as well.  

In discussion with @kybaker we felt that none of the icons in the masthead should use a caret next to them, and they should all behave as pop-ups rather than drop downs.  Also not clear on whether a background selected state is necessary for the pop-up, as that is normally used to indicate selection for a dropdown area instead, such as is used for the user dropdown or context selector dropdown.

image

@mcarrano - please review and provide your insights as well

### PR DESCRIPTION
**Pull request by deernano**
## Description
* This ia an update of general rules of masthead menu in masthead design pages. 

* Include related PRs, GitHub Issues and or Jira stories.
   https://github.com/patternfly/patternfly-design/issues/570#issuecomment-374966199

Closes #570 

## Changes

Write a list of changes this design pattern introduces

* 01 Changed the Overview page description and images.
* 02 Changed the Design page description and images.
* 03 Added new images of masthead menu.

## Contribution Submission Checklist

- [X] A GitHub issue was created and reviewed with the PatternFly community.
- [X] The pattern design overview.md and design.md templates were used to provide all information, relevant to this design.
- [X] Any new files have been added to the folder for that pattern.
- [ ] The folder, for brand new patterns, has been titled with the name of the pattern.
- [X] All images are in an "img" folder and embedded images have the correct filepath.
- [X] The overview.md, design.md and site.md files, for this design have been commited and pushed to the patternfly-design repository, for review.

### CODE REVIEW

**Comment by deernano** [comment]
@mcarrano @matthewcarleton Please help to review the PR. Thank you :)

File: pattern-library/application-framework/masthead/design/design.md [hunk 1]
```diff
@@ -2,48 +2,89 @@
...
+1.**Hamburger Icon:** The hamburger icon acts as a way for the user to open and close the vertical navigation panel. When using vertical navigation, we recommend always having the hamburger icon visible. When using horizontal navigation, the hamburger menu is only required on mobile devices.
 
+### Logotype
+2.**Logo** (optional): When using the tall masthead design, there is space for a product logo to be placed to the left of the product title for additional branding.
 
-## General Considerations
+3.**Application Title:** The logo area is in the left corner and contains the application product name image file, in the form of a sized SVG graphic.
+
+### Menu Label
+4.** Cuntext Selector:** The Context Selector is used when the data/resources shown in the UI are dependent on context. It is useful when the UI cannot show data for multiple contexts simultaneously, or when it makes sense not to. Jump to the [Context Selector](http://www.patternfly.org/pattern-library/navigation/context-selector/#overview) pattern for additional details.
```
Comment (matthewcarleton) [inline pattern-library/application-framework/masthead/design/design.md]: spelling "context"

File: pattern-library/application-framework/masthead/design/design.md [hunk 2]
```diff
@@ -2,48 +2,89 @@
...
+
+6.**Notifications Icon** (optional):
+  If notifications or messages are being used, the "fa-bell" icon should be placed to the left of the Application Launcher icon. The bell icon is used for opening and closing the notification drawer. Jump to the [Notification Drawer](https://www.patternfly.org/pattern-library/communication/notification-drawer/#/api) pattern for additional details.
+
+7.**Application Launcher Icon** (optional):
+   If additional interfaces are available via an app selection panel, the "fa-th" icon from the Font Awesome icon library should be placed to the left of the help icon. Jump to the [Launcher](http://www.patternfly.org/pattern-library/application-framework/launcher/#/api) pattern for additional details.
+
+8.**Help Icon:**
+  * The pficon-help icon should always be placed directly to the left of the user icon.
+  * The dropdown arrow should appear to the right of the icon to indicate additional options below.
```
Comment (matthewcarleton) [inline pattern-library/application-framework/masthead/design/design.md]: Am I mistaken or did I read that we were removing the arrow in this case? @mcarrano are you aware of this?

File: pattern-library/application-framework/masthead/design/design.md [hunk 3]
```diff
@@ -2,48 +2,89 @@
...
+  * The dropdown items may vary, but any product documentation or helpful website links available to users should be listed here. This would also be the appropriate location for customer support links.
+  * The dropdown should always include an "About" option that will launch the about modal for product version information. Jump to the [About Modal](https://www.patternfly.org/pattern-library/communication/about-modal/#/api) pattern for additional details.
 
-* **Badges:**
-  - In some cases, badges may be used with the icons in the masthead to provide the user with additional information.
+9.**User Icon:**
+  * The username of the current logged in user should be listed to the right of the pficon-user icon and should always be in the top right corner of the masthead.
+  * The dropdown arrow should appear to the right of the name to indicate additional options below.
```
Comment (matthewcarleton) [inline pattern-library/application-framework/masthead/design/design.md]: Again same with the arrow here.

File: pattern-library/application-framework/masthead/design/design.md [hunk 4]
```diff
@@ -2,48 +2,89 @@
...
+In smaller viewport sizes, the hamburger menu should list all primary navigation items, followed by the items from the masthead (except for badged items like notifications and shopping cart which should remain in the masthead if possible).
+
+Masthead items should be listed based on their order in the masthead from left to right. For instance, based on the example images, the hamburger menu would be the following from top to bottom: Category 1, Category 2, Category 3, Help, User.
+
+![Image of masthead details](img/masthead-humburger@2x.png)
+
+Any masthead options that have a dropdown should be displayed consistently with standard secondary navigation items.
+
+### Masthead Menu
+Masthead menu is a temporary box that appears when the user clicks a button, an icon or an icon with text on masthead. It contains a list of menu items. Each menu items initiate actions like opening a window, selecting an action or viewing information, etc.
```
Comment (matthewcarleton) [inline pattern-library/application-framework/masthead/design/design.md]: This should read "each menu item initiates actions like opening a window"

File: pattern-library/application-framework/masthead/design/design.md (hunk 2 above)
Comment (mcarrano) [reply pattern-library/application-framework/masthead/design/design.md]: Yes, I thought we were removing the arrows for all icon based items. Is that correct @deernano ?

File: pattern-library/application-framework/masthead/design/design.md (hunk 3 above)
Comment (mcarrano) [reply pattern-library/application-framework/masthead/design/design.md]: The user menu becomes more like a text menu with the user name there, so I think we're saying this _should_ have the arrow. Please confirm @deernano

File: pattern-library/application-framework/masthead/design/design.md [hunk 5]
```diff
@@ -2,48 +2,89 @@
...
+  * The dropdown arrow should appear to the right of the name to indicate additional options below.
+  * The dropdown items may vary based on the user settings and permissions available, but this section should always include a "Log out" option at the bottom of the dropdown list.
+  * If a [Language Selector](https://www.patternfly.org/pattern-library/forms-and-controls/language-selector/#/api) exists, it should appear in this dropdown menu as well.
+
+
+## General Considerations
+### Badges
+In some cases, badges may be used with the icons in the masthead to provide the user with additional information.
+![Image of masthead details](img/masthead-badge@2x.png)
   - Badges should be blue with white text inside and should be placed to the top right corner of the icon it is associated with.
```
Comment (mcarrano) [inline pattern-library/application-framework/masthead/design/design.md]: I think that we previously decided to not include text inside of these badges due to size constraints. Can you confirm @kybaker ?

File: pattern-library/application-framework/masthead/design/design.md [hunk 6]
```diff
@@ -2,48 +2,89 @@
...
   - Some examples include showing new notifications are available, or showing the number of items currently in a shopping cart.
   - In responsive states, most masthead icons will be collapsed into the responsive state, but any icons that use badges should remain in the masthead for easier access.
```
Comment (mcarrano) [inline pattern-library/application-framework/masthead/design/design.md]: I'm not really sure that we would be able to keep more that one icon in the masthead at mobile size. Can you comment @matthewcarleton ?

File: pattern-library/application-framework/masthead/design/design.md [hunk 7]
```diff
@@ -2,48 +2,89 @@
...
 
+### Hamburger Menu
+In smaller viewport sizes, the hamburger menu should list all primary navigation items, followed by the items from the masthead (except for badged items like notifications and shopping cart which should remain in the masthead if possible).
+
+Masthead items should be listed based on their order in the masthead from left to right. For instance, based on the example images, the hamburger menu would be the following from top to bottom: Category 1, Category 2, Category 3, Help, User.
+
```
Comment (mcarrano) [inline pattern-library/application-framework/masthead/design/design.md]: Add... "where Category 1, Category 2, and Category 3 are items from the primary navigation of the application.

File: pattern-library/application-framework/masthead/design/design.md [hunk 8]
```diff
@@ -2,48 +2,89 @@
...
+![Image of masthead details](img/masthead-humburger@2x.png)
```
Comment (mcarrano) [inline pattern-library/application-framework/masthead/design/design.md]: Help and User are flyout menus in this mode so they should have the right arrow.

File: pattern-library/application-framework/masthead/design/design.md [hunk 9]
```diff
@@ -2,48 +2,89 @@
...
+
+Any masthead options that have a dropdown should be displayed consistently with standard secondary navigation items.
+
+### Masthead Menu
```
Comment (mcarrano) [inline pattern-library/application-framework/masthead/design/design.md]: I think that the content in this section is good but it starts to become redundant with what we say above about menu items. Can we consolidate these two sections somehow to have one section that addresses menu items and how the work?

File: pattern-library/application-framework/masthead/design/design.md (hunk 2 above)
Comment (deernano) [reply pattern-library/application-framework/masthead/design/design.md]: @mcarrano @matthewcarleton yes we are removing all arrows.

**Comment by deernano** [comment]
@mcarrano @matthewcarleton  Please help to reivew the text in google doc so that we can make the next commit. Thank you！ https://docs.google.com/document/d/1K8vV18qKYUG-900xzUpiHeLtNvjmtGm3wg-4yrJqHtU/edit?usp=sharing

File: pattern-library/application-framework/masthead/design/design.md (hunk 9 above)
Comment (deernano) [reply pattern-library/application-framework/masthead/design/design.md]: @mcarrano I tried to delete some of the text in the above section. Maybe its still not good enough. Let's reivew it in google doc~

**Comment by mcarrano** [comment]
I reviewed your Google doc @deernano and made some suggestions to reduce the amount of text on the Design page. Let me know what you think.

File: pattern-library/application-framework/masthead/design/design.md (hunk 1 above)
Comment (deernano) [reply pattern-library/application-framework/masthead/design/design.md]: Updated

File: pattern-library/application-framework/masthead/design/design.md (hunk 3 above)
Comment (deernano) [reply pattern-library/application-framework/masthead/design/design.md]: Yes I agree with @mcarrano

**Comment by deernano** [comment]
@mcarrano @matthewcarleton Updated based on the comments and the google doc https://docs.google.com/document/d/1K8vV18qKYUG-900xzUpiHeLtNvjmtGm3wg-4yrJqHtU/edit?usp=sharing
Please review it. Thank you ~

**Comment by mcarrano** [comment]
@deernano This looks good now. Please just go ahead and update this PR with text and images and I will approve.

File: pattern-library/application-framework/masthead/design/design.md [hunk 10]
```diff
@@ -2,48 +2,82 @@
...
+
+#### Text
+If the item is a text or a button, use a caret on the right side of the text. Do not use text only.
+
+![Image of masthead details](img/masthead-menu3@2x.png)
+
+#### Icon with text
+If the item is an icon with text, use a caret on the right side of the icon.
+
+![Image of masthead details](img/masthead-menu4@2x.png)
```
[the remark this answers is not in the document]
Comment (deernano) [reply pattern-library/application-framework/masthead/design/design.md]: Updated

File: pattern-library/application-framework/masthead/design/design.md [hunk 11]
```diff
@@ -2,48 +2,82 @@
...
+![Image of masthead details](img/masthead-badge@2x.png)
   - Some examples include showing new notifications are available, or showing the number of items currently in a shopping cart.
   - In responsive states, most masthead icons will be collapsed into the responsive state, but any icons that use badges should remain in the masthead for easier access.
 
+### Hamburger Menu
+In smaller viewport sizes, the hamburger menu should list all primary navigation items, followed by the items from the masthead (except for badged items like notifications and shopping cart which should remain in the masthead if possible).
+
+Masthead items should be listed based on their order in the masthead from left to right. For instance, based on the example images, the hamburger menu would be the following from top to bottom: Category 1, Category 2, Category 3, Help, User. Where Category 1, Category 2, and Category 3 are items from the primary navigation of the application.
+
+![Image of masthead details](img/masthead-humburger@2x.png)
```
[the remark this answers is not in the document]
Comment (deernano) [reply pattern-library/application-framework/masthead/design/design.md]: Updated

**Pull request merged**